src/App.jsx

```diff
@@ -14,33 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_FIRST_LETTER = 'byLetter';
+const SORT_BY_GOOD_LENGTH = 'byLength';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getSortedGoods(goodsList, sortField, reversed) {
+  let goodsListCopy = [...goodsList];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    goodsListCopy.sort((good1, good2) => {
+      switch (sortField) {
+        case 'byLetter':
+          return good1.localeCompare(good2);
+        case 'byLength':
+          return good1.length - good2.length;
+        default:
+          return goodsListCopy;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reversed) {
+    goodsListCopy = [...goodsListCopy].reverse();
+  }
+
+  return goodsListCopy;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const renderedGoods = getSortedGoods(goodsFromServer, sortField, reversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': !(sortField === SORT_BY_FIRST_LETTER),
+          })}
+          onClick={() => setSortField(SORT_BY_FIRST_LETTER)}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': !(sortField === SORT_BY_GOOD_LENGTH),
+          })}
+          onClick={() => setSortField(SORT_BY_GOOD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => {
+            setReversed(!reversed);
+          }}
+        >
+          Reverse
+        </button>
+
+        {sortField === '' && !reversed ? (
+          ''
+        ) : (
```

Comment:
```suggestion
        {(sortField || reversed) && (
```

But create variable for `sortField || reversed` and use here